File: fizzbuzz.go
// Language:       go
// Web site:       http://golang.org/
// Last tested on: Ubuntu 18.04
// Requires:       apt-get install golang

// NOTE: The above currently installs Go version 1.6.1.
//       The latest version is 1.7.3, installable from
//       https://storage.googleapis.com/golang/go1.7.3.linux-amd64.tar.gz

package main

import "fmt"

func main() {
    for i := 1; i <= 100; i ++ {
        if i % 15 == 0 {
            fmt.Println("FizzBuzz")
        } else if i % 3 == 0 {
            fmt.Println("Fizz")
        } else if i % 5 == 0 {
            fmt.Println("Buzz")
        } else {
            fmt.Printf("%d\n", i)
        }
    }
}

File: fizzbuzz.zig
// Language:       Zig
// Web site:       https://ziglang.org/
// Last tested on: Ubuntu 22.04.3
// Requires:       sudo snap install --beta --classic zig
//                 (Installs zig 0.15.2)

const std = @import("std");

pub fn main() !void {
    const stdout = std.fs.File.stdout();
    var i: u32 = 1;
    while (i <= 100) : (i += 1) {
        if (i % 15 == 0) {
            try stdout.writeAll("FizzBuzz\n");
        }
        else if (i % 3 == 0) {
            try stdout.writeAll("Fizz\n");
        }
        else if (i % 5 == 0) {
            try stdout.writeAll("Buzz\n");
        }
        else {
            var buf: [3]u8 = undefined;
            const str = try std.fmt.bufPrint(&buf, "{}\n", .{i});
            try stdout.writeAll(str);
        }
    }
}
